papers/p1729r4.bs

```diff
@@ -1331,37 +1342,29 @@ Encoding {#encoding}
 
 In a similar manner as with `std::format`, input given to `std::scan` is assumed
 to be in the (ordinary/wide) literal encoding.
-Errors in encoding are handled in a "garbage in, garbage out" manner:
-invalidly encoded code points are treated as if they were the Unicode noncharacter U+FFFF,
-which doesn't match any other character or pattern.
 
-<div class=note>
-Encountering invalid encoding inside `std::scan` could be a use case
-for erroneous behavior in the library. That's because be provide well defined-behavior
-for handling invalid encoding, but it's still likely to be an error.
-As motivation for erroneous behavior,
-Unicode conformance requirement C.10 says that ill-formed input shall not be treated as a character,
-and treat it as an error instead.
-</div>
+If an error in encoding is encountered while reading a value of a string type
+(`basic_string`, `basic_string_view`), an `invalid_scanned_value` error is returned.
+For other types, the reading is stopped, as the parser can't parse a numeric value from
+something that isn't digits.
 
 <div class=example>
-```cpp
+```c++
 // Invalid UTF-8
 auto r = std::scan<std::string>("a\xc3 ", "{}");
-// r->value() == "a\xc3"
-// Erroneous behavior?
+// r == false
+// r->error() == std::scan_error::invalid_scanner_value
+
+auto r2 = std::scan<int>("1\xc3 ", "{}");
+// r2 == true
+// r2->value() == 1
+// r2->range() == "\xc3 "
 ```
 </div>
 
-Other potential options for handling invalid encoding would be:
-
- * treat is as UB
- * always sanitize input encoding (potentially very slow when done character-by-character with `forward_range`s)
- * check for encoding when reading code units and strings, while potentially introducing a format specifier for "raw mode",
-    which skips these checks
-
-Note: This topic is under active contention in SG16.
-See also example in [[#fmt-type-char]].
+Because `std::scan` is, as indicated by the paper title, a <i>text parsing</i> facility,
+raw bytes input into a `string` isn't supported. That can be achieved with simpler range algorithms
+already in the standard.
```

Comment:
I think it's OK as the default behavior but we might want to clarify that binary input is also possible. Otherwise the facility will be too limited (e.g. stdio, and `std::format` support binary).